mesheryctl/internal/cli/root/system/context.go

```diff
@@ -304,15 +304,30 @@ var switchContextCmd = &cobra.Command{
 		if viper.GetString("current-context") == args[0] {
 			return errors.New("already using context '" + args[0] + "'")
 		}
-		if err := stop(); err != nil {
-			return errors.Wrap(err, utils.SystemError("Failed to stop Meshery before switching context"))
+		//check if meshery is running
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return errors.Wrap(err, "error processing config")
+		}
+		currCtx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		isRunning, err := utils.IsMesheryRunning(currCtx.GetPlatform())
+		//if meshery running stop meshery before context switch
+		if isRunning {
+			if err := stop(); err != nil {
+				return errors.Wrap(err, utils.SystemError("Failed to stop Meshery before switching context"))
+			}
 		}
 		configuration.CurrentContext = args[0]
 		viper.Set("current-context", configuration.CurrentContext)
 		log.Printf("switched to context '%s'", args[0])
 		err = viper.WriteConfig()
-		if Starterr := start(); Starterr != nil {
-			return errors.Wrap(Starterr, utils.SystemError("Failed to start Meshery while switching context"))
+		if isRunning {
+			if Starterr := start(); Starterr != nil {
+				return errors.Wrap(Starterr, utils.SystemError("Failed to start Meshery while switching context"))
+			}
 		}
```

Comment:
this will never start, as meshery is not running in this case, you just stopped it above.

Reply:
It only starts meshery if it was running earlier.

Do we need to start meshery on context switch irrespective of it is running or not?

Reply:
ooh, my mistake. I misread it. It's good!